### PR TITLE
feat(calls): call improvements & group call

### PR DESCRIPTION
**Pull request by aewing**
**What this PR does** 📖
Adds screen sharing to calls & makes some changes to how peers and calls are managed to set the stage for group calls and hopefully improve call user experience.

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by phillsatellite** [comment]
I noticed this error message when User accepts a friend request I couldn't replicate this on dev branch or test.satellite.one

Screen Shot 2022-05-05 at 3 28 01 PM

**Review by stavares843 — CHANGES_REQUESTED** [review]
I also have the same error as Phil mentioned above

also,

- if we don't allow permissions to share screen, will crash when trying to do a share screen

Captura de ecrã 2022-05-05, às 22 11 58

- when trying again, still crash's but doesn't ask again the permission it just crash's again


https://user-images.githubusercontent.com/29093946/167027497-94822aa8-19c3-4489-8d4b-5ef810c17d74.mov


- when trying to do a call with a user that is not online, the video call icon is still visible and console log returns `call - conversation not initialized or id not found`

Captura de ecrã 2022-05-05, às 22 13 22

- if you are on a chat with 1 person and you also have a chat with a group, if you are on the DM and you get a group DM, even if you accept the call you are not redirected for the call, you stay on the DM

Captura de ecrã 2022-05-05, às 22 27 20
Captura de ecrã 2022-05-05, às 22 27 45
Captura de ecrã 2022-05-05, às 22 27 57

- when you get a group DM, all is black on the modal, doesn't say who is calling

Captura de ecrã 2022-05-05, às 22 27 14

- even when the video call icon is not available, user is still able to proceed to call the group 

https://user-images.githubusercontent.com/29093946/167029823-08671e2e-9152-4b9c-a05b-5400e3fcb9dd.mov


- when user A starts group call and user B accepts call, there are some grey cubes without name on the call

Captura de ecrã 2022-05-05, às 22 36 20

- i'm already on a call and then without anyone calling because the call was just with users locally, the call modal was appearing anyway

https://user-images.githubusercontent.com/29093946/167030230-5791b16f-5d06-4327-b7f5-9824735603ae.mov

**Comment by phillsatellite** [comment]
I also found a few more bugs on mobile (iOS) 
- If you add a iOS user to a group, that group will never appear in the sidebar
IMG_5328
Screen Shot 2022-05-05 at 6 36 25 PM

- Calling someone with screen-share call will never appear on the other user's side (desktop as well)
- on IOS doesn't ask what screen you want to share just immediately starts calling (call will never go through to other user)

https://user-images.githubusercontent.com/93608357/167036914-3ae73e6b-c567-42b6-b93b-5da99278df8b.MP4

**Comment by phillsatellite** [comment]
A few things I found after testing that I wasnt able to replicate on dev branch

1. I wasn't able to ever get screensharing to work. I would click which screen to share then nothing happens 
https://user-images.githubusercontent.com/93608357/167913216-05191620-ead9-4fc3-9219-df2b8f11d02a.mov

2. I noticed when you load into a chat the textbox and calling tools at the top will be missing. User has to navigate to a different page and back to the chat for it to appear 
https://user-images.githubusercontent.com/93608357/167913514-fcce3f4f-76d2-4025-91ce-ec139f4a3354.mov

3.The group chat screens overlapped, I tried this on dev branch but couldnt get 3 people in a group
group chat overlap
 
4. After ending group chat call input box would be in the middle of the screen (this would persist to other chats as well)
input box

**Comment by Clayton215** [comment]
Upon sending pictures in group messages the app crashes and shows this error.
The picture I used in this instance was a .PNG however the crash also happened with different types such as .jpegs and .heic

https://user-images.githubusercontent.com/96502187/168117305-b01953db-7859-4f93-a3df-b3f70ed61fec.mov
console log

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/168136253-68b10675-d726-4908-8a3e-3ae55d7b8368.mov


https://user-images.githubusercontent.com/29093946/168136286-2b448bbf-225b-49a6-af32-716774efec71.mov

if i'm not on a call, trying to use the toggles crash's

m1, chrome

**Review by stavares843 — CHANGES_REQUESTED** [review]
user A called user B

user B didn't get the call

user B called user A

user A declined the call and also closed the call on the user B's side

**Comment by aewing** [comment]
> user A called user B
> 
> user B didn't get the call
> 
> user B called user A
> 
> user A declined the call and also closed the call on the user B's side

@stavares843 if User A calls User B before user B has a chance to establish a connection to User A, the call will not be delivered right away, but should dial through after the connection is established. Is this not the case? Could you maybe make a recording?

**Review by stavares843 — CHANGES_REQUESTED** [review]
Captura de ecrã 2022-05-12, às 18 55 39

hang up a call, the modal appeared like this

not always reproducible

brave, m1

**Review by stavares843 — CHANGES_REQUESTED** [review]
Captura de ecrã 2022-05-12, às 18 56 41

made a call and the modal was there if you see the glyph is cut off with the same size the call modal has but no call modal displaying

can't always reproduce

brave, m1

**Review by stavares843 — CHANGES_REQUESTED** [review]
Captura de ecrã 2022-05-12, às 19 07 18

this branch vs dev

user is not connected on the bottom is not showing on this branch

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/168140782-1611d8b1-3542-4f7f-8dd5-b7b87241445b.mov

i can only sends glyphs if the user is online

on dev works fine

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/168143452-2f14b184-d045-46f8-9252-2c59629c994e.mov

the icon is deactivated but is still clickable same with the call one

brave, m1

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/168143781-230b4be7-f58b-4496-ad9c-424febf14ea2.mov


doing a call group is not appearing on both sides even after almost 1m

brave and opera, m1

**Review by WanderingHogan — APPROVED** [review]

File: store/friends/mutations.ts [hunk 1]
```diff
@@ -49,11 +49,11 @@ const mutations = {
...
   setActive(state: FriendsState, friend: Friend) {
-    state.all = state.all.map((f) => ({
-      ...f,
-      activeChat: f.address === friend.address,
-    }))
+    state.activeConversation = { type: 'friend', target: friend }
   },
+  // setActiveGroup(state: FriendsState, groupId: string) {
+  //   state.activeGroup = groupId
+  // },
```
Comment (stavares843) [inline store/friends/mutations.ts:56]: do we need this commented out?

**Comment by stavares843** [comment]
as discussed with the team, we will get this merged and add tickets for the bugs that this PR is introducing to be fixed separately

**Pull request merged**